gdal/configure.ac

```diff
@@ -1689,6 +1689,77 @@ fi
 
 AC_SUBST(ZSTD_SETTING,$ZSTD_SETTING)
 
+
+dnl ---------------------------------------------------------------------------
+dnl Check if blosc is available.
+dnl ---------------------------------------------------------------------------
+
+AC_ARG_WITH(blosc,[  --with-blosc[=ARG]       Include blosc support (ARG=yes/no/installation_prefix)],,)
+
+if test "$with_blosc" = "" -o "$with_blosc" = "yes" ; then
+  AC_CHECK_LIB(blosc,blosc_cbuffer_validate,HAVE_BLOSC=yes,HAVE_BLOSC=no,)
+
+  if test "$HAVE_BLOSC" = "yes" ; then
+    LIBS="-lblosc $LIBS"
+  else
+    if test "$with_blosc" = "yes" ; then
+      AC_MSG_ERROR([libblosc not found])
+    else
+      echo "libblosc not found - BLOSC support disabled"
+    fi
+  fi
+elif test "$with_blosc" != "" -a "$with_blosc" != "no"; then
+
+  AC_CHECK_LIB(blosc,blosc_cbuffer_validate,HAVE_BLOSC=yes,HAVE_BLOSC=no,-L$with_blosc/lib)
+
+  if test "$HAVE_BLOSC" = "yes" -a -f "$with_blosc/include/blosc.h" ; then
+    LIBS="-L$with_blosc/lib -lblosc $LIBS"
+    EXTRA_INCLUDES="-I$with_blosc/include $EXTRA_INCLUDES"
+  else
+    AC_MSG_ERROR([libblosc not found])
+  fi
+
+else
+    HAVE_BLOSC=no
+fi
+
+AC_SUBST(HAVE_BLOSC,$HAVE_BLOSC)
```

Comment:
Too fragile, c-blosc may depend on lz4, snappy, zlib and zstd. A naive `AC_CHECK_LIB` may break due to missing dependencies in LIBS. c-blosc provides a pkg-config file: `blosc.pc`

Reply:
> Too fragile, c-blosc may depend on lz4, snappy, zlib and zstd. A naive `AC_CHECK_LIB` may break due to missing dependencies in LIBS. c-blosc provides a pkg-config file: `blosc.pc`

too fragile for static builds of c-blosc ? should work fine hopefully for dynamic builds of c-blosc. Anyway autoconf builds will be soon deprecated, so any improvements if any needed should go to the CMake builds

Reply:
> too fragile for static builds of c-blosc ?

yes (in conan-center, by default all libs are built against static dependencies so we quickly see these kind of issues).

> Anyway autoconf builds will be soon deprecated, so any improvements if any needed should go to the CMake builds

Great ! I'll check that while packaging the first release using CMakeLists.

Reply:
> Great ! I'll check that while packaging the first release using CMakeLists.

early testing of CMake builds of master would be appreciated